### ISSUE TITLE
Error when targeting with grains via REST API

### ISSUE THREAD

**Issue by clallen** [saltstack/salt#54476]
### Description of Issue
I am getting an exception in the master log when trying to target with the "grains" target type, using the REST API:
```
2019-09-13 16:49:25,500 [salt.utils.minions                                              :696 ][ERROR   ][39] Failed matching available minions with grains pattern: group:1
Traceback (most recent call last):
  File "/salt-2018.3.3/lib/python2.7/site-packages/salt/utils/minions.py", line 685, in check_minions
    _res = check_func(expr, greedy)
TypeError: 'NoneType' object is not callable
2019-09-13 16:49:25,503 [salt.utils.data                                                 :572 ][DEBUG   ][39] Attempting to match '1' in 'group' using delimiter ':'
2019-09-13 16:49:25,505 [salt.utils.data                                                 :572 ][DEBUG   ][39] Attempting to match '1' in 'group' using delimiter ':'
2019-09-13 16:49:25,506 [salt.utils.data                                                 :572 ][DEBUG   ][39] Attempting to match '1' in 'group' using delimiter ':'
2019-09-13 16:49:25,508 [salt.utils.data                                                 :572 ][DEBUG   ][39] Attempting to match '1' in 'group' using delimiter ':'
```
Using the "G@" compound target type works.

### Setup
Docker test environment with 2 masters and 4 minions.  
2 of the minions have the "group:1" grain set.

Master API config
```
auth.ldap.activedirectory: True
auth.ldap.filter: sAMAccountName={{username}}
auth.ldap.accountattributename: sAMAccountName
auth.ldap.groupclass: group
auth.ldap.persontype: person
auth.ldap.server: [redacted]
auth.ldap.port: 3268
auth.ldap.binddn: [redacted]
auth.ldap.bindpw: [redacted]
external_auth:
  ldap:
    Unix Admins%:
      - 'G@group:1':
        - test.ping
```

### Steps to Reproduce Issue
1. From one of the minions, authenticate to the master:
```
curl -Ssik https://salt-master-1:8000/login -c /tmp/cookies -H 'Accept: application/x-yaml' -d username=[redacted] -d password=[redacted] -d eauth=ldap
```
Output:
```
HTTP/1.1 200 OK
Content-Length: 183
Access-Control-Expose-Headers: GET, POST
Vary: Accept-Encoding
Server: CherryPy/17.4.2
Allow: GET, HEAD, POST
Access-Control-Allow-Credentials: true
Date: Fri, 13 Sep 2019 16:30:15 GMT
Access-Control-Allow-Origin: *
X-Auth-Token: c3c4ec94f7b1af80656ee981fb65d484557d760e
Content-Type: application/x-yaml
Set-Cookie: session_id=c3c4ec94f7b1af80656ee981fb65d484557d760e; expires=Sat, 14 Sep 2019 02:30:15 GMT; Max-Age=36000; Path=/

return:
- eauth: ldap
  expire: 1568435415.386854
  perms:
  - G@group:1:
    - test.ping
  start: 1568392215.386853
  token: c3c4ec94f7b1af80656ee981fb65d484557d760e
  user: [redacted]
```

2. Send a test.ping command to the minions with "group:1" grain target:
```
curl -Ssik https://salt-master-1:8000 -b /tmp/cookies -H 'Accept: application/x-yaml' -d client=local -d tgt='group:1' -d tgt_type=grain -d fun=test.ping
```
Output:
```
HTTP/1.1 200 OK
Content-Length: 13
Access-Control-Expose-Headers: GET, POST
Cache-Control: private
Vary: Accept-Encoding
Server: CherryPy/17.4.2
Allow: GET, HEAD, POST
Access-Control-Allow-Credentials: true
Date: Fri, 13 Sep 2019 16:49:25 GMT
Access-Control-Allow-Origin: *
Content-Type: application/x-yaml
Set-Cookie: session_id=c3c4ec94f7b1af80656ee981fb65d484557d760e; expires=Sat, 14 Sep 2019 02:49:25 GMT; Max-Age=36000; Path=/

return:
- {}
```
This is where I see the above exception in the master log.

However, using the "compound" target type works:
```
[root@14f69b9d9988 /]# curl -Ssik https://salt-master-1:8000 -b /tmp/cookies -H 'Accept: application/x-yaml' -d client=local -d tgt='G@group:1' -d tgt_type=compound -d fun=test.ping
HTTP/1.1 200 OK
Content-Length: 50
Access-Control-Expose-Headers: GET, POST
Cache-Control: private
Vary: Accept-Encoding
Server: CherryPy/17.4.2
Allow: GET, HEAD, POST
Access-Control-Allow-Credentials: true
Date: Fri, 13 Sep 2019 16:52:06 GMT
Access-Control-Allow-Origin: *
Content-Type: application/x-yaml
Set-Cookie: session_id=c3c4ec94f7b1af80656ee981fb65d484557d760e; expires=Sat, 14 Sep 2019 02:52:06 GMT; Max-Age=36000; Path=/

return:
- 14f69b9d9988: true
  551ce2cc57fe: true
```
Running it on the master's command line works as well:
```
[root@91a2459731a6 tmp]# salt -a ldap --username=[redacted] --password=[redacted] -G group:1 test.ping
551ce2cc57fe:
    True
14f69b9d9988:
    True
```

### Versions Report
```
Salt Version:
           Salt: 2018.3.3

Dependency Versions:
           cffi: Not Installed
       cherrypy: 17.4.2
       dateutil: 2.8.0
      docker-py: 4.0.2
          gitdb: 2.0.5
      gitpython: 2.1.14
          ioflo: Not Installed
         Jinja2: 2.10.1
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.6.1
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.5 (default, Jun 20 2019, 13:30:36)
   python-gnupg: Not Installed
         PyYAML: 5.1.2
          PyZMQ: 17.0.0
           RAET: Not Installed
          smmap: 2.0.5
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.1.6

System Versions:
           dist: oracle 7.6
         locale: ANSI_X3.4-1968
        machine: x86_64
        release: 4.14.35-1844.4.5.2.el7uek.x86_64
         system: Linux
        version: Oracle Linux Server 7.6
```

**Comment by xeacott** [comment]
Not stale.

**Comment by xeacott** [comment]
Is there a way you could test to see if this is still occurring in versions `2019.2` and above?

**Comment by clallen** [comment]
Sure, I should be able to get to it in the next week or so.

**Comment by xeacott** [comment]
Thank you!

**Comment by clallen** [comment]
Looks to be fixed in 2019.2.2, I was able to use the grain and compound target types.  I'll close this now.  Thanks!

```
[root@7944c96bbb35 /]# curl -Ssik https://salt-master-1:8000 -b /tmp/cookies -H 'Accept: application/x-yaml' -d client=local -d tgt='group:1' -d tgt_type=grain -d fun=test.ping
HTTP/1.1 200 OK
Content-Length: 50
Access-Control-Expose-Headers: GET, POST
Cache-Control: private
Vary: Accept-Encoding
Server: CherryPy/17.4.2
Allow: GET, HEAD, POST
Access-Control-Allow-Credentials: true
Date: Mon, 03 Feb 2020 15:57:14 GMT
Access-Control-Allow-Origin: *
Content-Type: application/x-yaml
Set-Cookie: session_id=f83fbee8231a9334e85029e2d6f0325f47419920; expires=Tue, 04 Feb 2020 01:57:14 GMT; Max-Age=36000; Path=/

return:
- 03b9f5b70620: true
  83d470ec1508: true
[root@7944c96bbb35 /]# curl -Ssik https://salt-master-1:8000 -b /tmp/cookies -H 'Accept: application/x-yaml' -d client=local -d tgt='G@group:1' -d tgt_type=compound -d fun=test.ping
HTTP/1.1 200 OK
Content-Length: 50
Access-Control-Expose-Headers: GET, POST
Cache-Control: private
Vary: Accept-Encoding
Server: CherryPy/17.4.2
Allow: GET, HEAD, POST
Access-Control-Allow-Credentials: true
Date: Mon, 03 Feb 2020 16:02:35 GMT
Access-Control-Allow-Origin: *
Content-Type: application/x-yaml
Set-Cookie: session_id=f83fbee8231a9334e85029e2d6f0325f47419920; expires=Tue, 04 Feb 2020 02:02:35 GMT; Max-Age=36000; Path=/

return:
- 03b9f5b70620: true
  83d470ec1508: true
```
